src/metrics/metrics.ts

```diff
@@ -326,49 +342,107 @@ function checkOpening() {}
-// the edge case here is when pieces "share" a mate, or check. This can be at most 2 due to discovery checks
-// the board configuration will also have to be as it was in the instance of checkmate, or the previous mate.
-// So this is going to need to hook into the loadPGN, probably.
-export function getMateAndAssists(gameHistory: GameHistoryMove[]) {
-  let assistingPiece;
+// DONE
+// One edge case currently unaccounted for is when pieces "share" a mate, or check. This can be at most 2 due to discovery checks (currently we disregard this by just referring to whatever the PGN says. If the piece that moves causes checkmate, then it is the "mating piece")
+export function getMateAndAssists(pgnMoveLine: string) {
+  const chess = new Chess();
+  const moveGenerator = chess.historyGenerator(pgnMoveLine);
+
   let matingPiece;
+  let assistingPiece;
   let hockeyAssist;
+  let unambigMatingPiece;
+  let unambigMatedPiece;
+  let unambigAssistingPiece;
+  let unambigHockeyAssistPiece;
+  let lastPieceMoved;
+  let lastMove: PrettyMove | null = null;
 
-  // check for mate
-  if (gameHistory[gameHistory.length - 1].originalString.includes('#')) {
-    matingPiece = gameHistory[gameHistory.length - 1].piece; // this doesn't disambiguate to the starting square of the piece; we'd want a chess.js rewrite to do that.
-
-    // If mate see if also assist
-    const assistCandidate = gameHistory[gameHistory.length - 3].piece;
-    if (
-      gameHistory[gameHistory.length - 3].originalString.includes('+') &&
-      matingPiece !== assistCandidate
-    ) {
-      assistingPiece = assistCandidate;
-
-      // If assist check for hockey assist
-      const hockeyCandidate = gameHistory[gameHistory.length - 5].piece;
-      if (
-        gameHistory[gameHistory.length - 5].originalString.includes('+') &&
-        assistingPiece !== hockeyCandidate &&
-        matingPiece !== hockeyCandidate
-      ) {
-        hockeyAssist = hockeyCandidate;
-      }
+  // Keep track of the last few moves
+  let lastFewMoves: PrettyMove[] = [];
+
+  for (let moveInfo of moveGenerator) {
+    const { move, board } = moveInfo;
+
+    lastMove = move;
+
+    // Add the current move to the start of the array
+    lastFewMoves.unshift(move);
+
+    // If we have more than 5 moves in the array, remove the oldest one
+    if (lastFewMoves.length > 5) {
+      lastFewMoves.pop();
     }
+
   }
+  console.log("last move: ", lastMove)
+
+  if (lastMove) {
+    lastPieceMoved = lastMove.unambiguousSymbol;
+
+    if (lastMove.originalString.includes('#')) {
+      matingPiece = lastMove.piece;
+      unambigMatingPiece = lastMove.unambiguousSymbol;
 
-  // This is where we DO need to disambiguate, the same piece type but a different piece could provide the assist.
-  // Pausing further work till we've fixed that
+      // Determine the color of the mated king
+      const matedKingColor = lastMove.color === 'w' ? 'b' : 'w';
+      unambigMatedPiece = matedKingColor === 'w' ? 'K' : 'k';
```

Comment:
I am confused why we need all of these variables, maybe we only need the last one?

Reply:
unambigMatingPiece used in KDR function to update kill count for the mating piece
matingPiece defined just to have if we want ambig mating piece kdrs down the line
matedKingColor just used to identify which king was mated, which is then used in KDR funct to update death count for the mated king